### PR TITLE
[24]: IoC implementation

### PR DESCRIPTION
**Pull request by IvanShetkoGodel**
Added autofac packages and AutofacConfig class

### CODE REVIEW

File: GodelMastery.FleaMarket/GodelMastery.FleaMarket.Web/App_Start/AutofacConfig.cs [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+using Autofac.Integration.Mvc;
+using IContainer = Autofac.IContainer;
+using System.Reflection;
+using System.Web.Mvc;
+
+namespace GodelMastery.FleaMarket.Web.App_Start
+{
+    public class AutofacConfig
+    {
+        public static void ConfigureContainer()
```
Comment (vladkuzmich) [inline GodelMastery.FleaMarket/GodelMastery.FleaMarket.Web/App_Start/AutofacConfig.cs]: Make class static please

File: GodelMastery.FleaMarket/GodelMastery.FleaMarket.Web/App_Start/AutofacConfig.cs [hunk 2]
```diff
@@ -0,0 +1,20 @@
+﻿using Autofac;
+using Autofac.Integration.Mvc;
+using IContainer = Autofac.IContainer;
```
Comment (Zhunusau) [inline GodelMastery.FleaMarket/GodelMastery.FleaMarket.Web/App_Start/AutofacConfig.cs]: Why do you need this here? Where do you use this?

**Comment by Zhunusau** [comment]
Where will be compare dependencies, in this class or in a separate Module ? Did you checked, it worked ?Example,
builder.RegisterType<TestClass>().As<ITestClass>().InstancePerRequest();

File: GodelMastery.FleaMarket/GodelMastery.FleaMarket.Web/App_Start/AutofacConfig.cs [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+
+namespace GodelMastery.FleaMarket.Web.App_Start
+{
+    public static class AutofacConfig
+    {
+        public static void ConfigureContainer()
+        {
+            var builder = new ContainerBuilder();
+
+            builder.RegisterControllers(typeof(MvcApplication).Assembly);
```
Comment (kpavaliayeu) [inline GodelMastery.FleaMarket/GodelMastery.FleaMarket.Web/App_Start/AutofacConfig.cs]: Move registration controllers into ControllerModule or remove ControllerModule.

**Pull request merged**